pkg/provisioner/ironic/ironic.go

```diff
@@ -456,6 +456,20 @@ func (p *ironicProvisioner) ValidateManagementAccess(data provisioner.Management
 		// below.
 	}
 
+	// If no PreprovisioningImage builder is enabled we set the Node network_data
+	// this enables Ironic to inject the network_data into the ramdisk image
+	if !p.config.havePreprovImgBuilder {
+		p.log.Info("adding preprovisioning network_data for node", "node", ironicNode.UUID)
+		networkDataRaw := data.PreprovisioningNetworkData
+		if networkDataRaw != "" {
+			var networkData map[string]interface{}
```

Comment:
nit: you could probably use `json.RawMessage` since we don't care about the exact format